packages/payload-auth/src/better-auth/plugin/lib/build-collections/index.ts

```diff
@@ -18,66 +18,55 @@ import { buildOauthConsentsCollection } from './oauth-consents'
 import { buildPasskeysCollection } from './passkeys'
 import { buildSsoProvidersCollection } from './sso-providers'
 import { buildAdminInvitationsCollection } from './admin-invitations'
+import { buildSubscriptionsCollection } from './subscriptions'
 
 /**
  * Builds the required collections based on the BetterAuth options and plugins
  */
-export function buildCollections({
+export function buildCollectionMap({
   incomingCollections,
   requiredCollectionSlugs,
-  pluginOptions,
-  betterAuthOptions
+  pluginOptions
 }: {
   incomingCollections: CollectionConfig[]
-  requiredCollectionSlugs: Set<string>
+  requiredCollectionSlugs: string[]
   pluginOptions: BetterAuthPluginOptions
-  betterAuthOptions: SanitizedBetterAuthOptions
-}): CollectionConfig[] {
+}): Record<string, CollectionConfig> {
   const buildCollectionMap = {
-    [baseCollectionSlugs.users]: () =>
-      buildUsersCollection({
-        incomingCollections,
-        betterAuthOptions,
-        pluginOptions
-      }),
-    [baseCollectionSlugs.accounts]: () => buildAccountsCollection({ incomingCollections, pluginOptions }),
-    [baseCollectionSlugs.sessions]: () =>
-      buildSessionsCollection({
-        incomingCollections,
-        betterAuthOptions,
-        pluginOptions
-      }),
-    [baseCollectionSlugs.verifications]: () => buildVerificationsCollection({ incomingCollections, pluginOptions }),
-    [baseCollectionSlugs.adminInvitations]: () =>
-      buildAdminInvitationsCollection({
-        incomingCollections,
-        pluginOptions
-      }),
-    [betterAuthPluginSlugs.organizations]: () => buildOrganizationsCollection({ pluginOptions }),
-    [betterAuthPluginSlugs.members]: () => buildMembersCollection({ pluginOptions }),
-    [betterAuthPluginSlugs.invitations]: () => buildInvitationsCollection({ pluginOptions }),
-    [betterAuthPluginSlugs.teams]: () => buildTeamsCollection({ pluginOptions }),
-    [betterAuthPluginSlugs.jwks]: () => buildJwksCollection({ pluginOptions }),
-    [betterAuthPluginSlugs.apiKeys]: () => buildApiKeysCollection({ pluginOptions }),
-    [betterAuthPluginSlugs.twoFactors]: () => buildTwoFactorsCollection({ pluginOptions }),
-    [betterAuthPluginSlugs.oauthAccessTokens]: () => buildOauthAccessTokensCollection({ pluginOptions }),
-    [betterAuthPluginSlugs.oauthApplications]: () => buildOauthApplicationsCollection({ pluginOptions }),
-    [betterAuthPluginSlugs.oauthConsents]: () => buildOauthConsentsCollection({ pluginOptions }),
-    [betterAuthPluginSlugs.passkeys]: () => buildPasskeysCollection({ pluginOptions }),
-    [betterAuthPluginSlugs.ssoProviders]: () => buildSsoProvidersCollection({ pluginOptions })
+    [baModelKeyToSlug.user]: () => buildUsersCollection({ incomingCollections, pluginOptions, collectionMap }),
```

Comment:
`collectionMap` references before actually have any values. This should not work/throw error

Reply:
After taking a closer look, it seems this function is used to get the collection slug from the Better Auth model key. We probably need to introduce a function that doesn’t require all collections to be built first, as that’s causing issues. Instead, we could have a simpler function that just takes pluginOptions and a model key, then returns the corresponding collection slug.